core/src/main/scala-3/pureconfig/generic/derivation/EnumConfigConvertDerivation.scala

```diff
@@ -0,0 +1,36 @@
+package pureconfig
+package generic
+package derivation
+
+import scala.compiletime.{constValue, erasedValue, error, summonInline}
+import scala.deriving.Mirror
+
+import com.typesafe.config.ConfigValue
+
+import pureconfig.error.{CannotConvert, ConfigReaderFailures}
+import pureconfig.generic.derivation.Utils.*
+
+type EnumConfigConvert[A] = EnumConfigConvertDerivation.Default.EnumConfigConvert[A]
+
+trait EnumConfigConvertDerivation extends EnumConfigReaderDerivation, EnumConfigWriterDerivation {
+  trait EnumConfigConvert[A] extends ConfigConvert[A]
```

Comment:
As I mentioned in the other PR, I'm not very fond of maintaining lots of logic for `ConfigConvert`, even if it's just boilerplate. `ConfigConvert` is meant to be a very thin class on top of `ConfigReader` and `ConfigWriter`...

I guess my first question is: do we really need to provide users the convenience of writing `derives ConfigConvert` instead of `derives ConfigReader, ConfigReader`? Is is worth having additional non-trivial (i.e. not just a one liner calling `ConfigConvert.fromReaderAndWriter`) derivation code here? I'd like to hear your thoughts too, @jcazevedo.

If we really want to be able to write `derives ConfigConvert`, does it need to have a class structure as complex as the other derivation cake patterns? Can `derived` implementations summon instances for other classes using the same `derived` mechanism? (searching imports for the appropriate `derived` implementation rather than statically choosing the inline function to call)

Reply:
> do we really need to provide users the convenience of writing derives ConfigConvert instead of derives ConfigReader, ConfigReader

I think that's useful to have.

> If we really want to be able to write derives ConfigConvert, does it need to have a class structure as complex as the other derivation cake patterns?

Maybe not 🤔 
I'll try experimenting with it a little bit more.

Reply:
Looks nice now! We can't avoid having a separate type but at least it's a straightforward `derived` implementation rather than a web of traits.